File: Reflow/Tasks/DataFlowTask.cs
using Reflow.Transformation;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;

namespace Reflow.Tasks
{
    public class DataFlowTask : TaskBase
    {
        public ILinkReader Input { get; set; }
        public ILinkWriter Output { get; set; }
        public ColumnMappings Mapping { get; set; }
        public bool IsAutoMap { get; set; }
        public string TableName { get; set; }

        public DataFlowTask()
        {
            this.Mapping = new ColumnMappings();
        }

        public override TaskResult OnExecute(TaskExecutionContext context)
        {
            Input.Open();
            TransformedReader reader = new TransformedReader(Input.Reader);
            reader.ScriptingLanguage = "VBScript";
            reader.TransformationScript = BuildScript();
            reader.Init();
            Output.Write(reader, TableName, context);
            return null;
        }

        private string BuildScript()
        {
            StringBuilder sb = new StringBuilder();
            foreach (ColumnMap map in this.Mapping)
            {
                if (map.SourceColumn == map.Destination)
                {
                    // skip;
                }
                else if (map.TransformExpression!=null)
                {
                    if (!string.IsNullOrEmpty(map.Destination))
                    {
                        sb.AppendLine(string.Format("Record.Item(\"{0}\") = {1}", map.Destination, map.TransformExpression.Code));
                    }
                }
            }
            return sb.ToString();
        }
    }
}

File: Java/src/main/java/com/nuarca/etl/tasks/DataFlowTask.java
//
// Translated by CS2J (http://www.cs2j.com): 12/6/2017 2:34:22 PM
//

package com.nuarca.etl.tasks;

import CS2JNet.System.StringSupport;
import com.nuarca.etl.ColumnMap;
import com.nuarca.etl.ColumnMappings;
import com.nuarca.etl.provider.ILinkReader;
import com.nuarca.etl.provider.ILinkWriter;
import com.nuarca.etl.tasks.TaskBase;
import com.nuarca.etl.tasks.TaskExecutionContext;
import com.nuarca.etl.tasks.TaskResult;
import com.nuarca.etl.transformation.TransformedReader;

public class DataFlowTask  extends TaskBase 
{
    private ILinkReader __Input;
    public ILinkReader getInput() {
        return __Input;
    }

    public void setInput(ILinkReader value) {
        __Input = value;
    }

    private ILinkWriter __Output;
    public ILinkWriter getOutput() {
        return __Output;
    }

    public void setOutput(ILinkWriter value) {
        __Output = value;
    }

    private ColumnMappings __Mapping;
    public ColumnMappings getMapping() {
        return __Mapping;
    }

    public void setMapping(ColumnMappings value) {
        __Mapping = value;
    }

    private boolean __IsAutoMap;
    public boolean getIsAutoMap() {
        return __IsAutoMap;
    }

    public void setIsAutoMap(boolean value) {
        __IsAutoMap = value;
    }

    private String __TableName;
    public String getTableName() {
        return __TableName;
    }

    public void setTableName(String value) {
        __TableName = value;
    }

    public DataFlowTask() throws Exception {
        this.setMapping(new ColumnMappings());
    }

    public TaskResult onExecute(TaskExecutionContext context) throws Exception {
        getInput().open();
        TransformedReader reader = new TransformedReader(getInput().getReader());
        reader.setMaps(this.getMapping());
        reader.setScriptingLanguage("VBScript");
        reader.setTransformationScript(buildScript());
        reader.init();
        getOutput().write(reader,getTableName(),context);
        return null;
    }

    private String buildScript() throws Exception {
        StringBuilder sb = new StringBuilder();
        for (Object __dummyForeachVar0 : this.getMapping())
        {
            ColumnMap map = (ColumnMap)__dummyForeachVar0;
            if (StringSupport.equals(map.getSourceColumn(), map.getDestination()))
            {
            }
            else // skip;
            if (map.getTransformExpression() != null)
            {
                if (!StringSupport.isNullOrEmpty(map.getDestination()))
                {
                    sb.append((String.format(StringSupport.CSFmtStrToJFmtStr("Record.Item(\"{0}\") = {1}"),map.getDestination(),map.getTransformExpression().getCode())) + System.getProperty("line.separator"));
                }
                 
            }
              
        }
        return sb.toString();
    }

}


